tscat/__init__.py

```diff
@@ -320,16 +320,14 @@ def export_json(catalogue: Catalogue) -> str:
 
 
 @typechecked
-def import_json(jsons: str) -> None:
+def canonicalize_json_import(jsons: str) -> dict:
     import_dict = json.loads(jsons)
 
     # check events and catalogues for existing entities
     # if existing and identical - remove from import-dict
     # if existing and not identical raise
     # if not existing import
 
-    event_of_uuid = {}
-
     for event in import_dict['events'][:]:
         check_event = get_events(UUIDFilter(event['uuid']))
```

Comment:
calling this for each event isn't much slower than calling once for all events?

Reply:
You mean by creating an Any() predicate of all event's UUIDs? Maybe.